### ISSUE TITLE
`:new edit` doesn't work with filenames containing spaces

### ISSUE THREAD

**Issue by gnull** [mawww/kakoune#1432]
Consider a directory containing the following file:

```
$ touch 'a b'
$ ls
'a b'
```

I launch kakoune in this directory without any arguments, start opening the file in a new window: `:new edit`, hit `<tab>`, it completes filename to `:new edit a\ b`, and I hit enter. The new widow is created but instead of displaying file contents it shows an error message:

```
1:1: 'edit' b is not a number
```

Running `:new edit 'a b'` returns the same error message. At the same time, if I try to edit the file in current window `:edit a\ b`, it works as expected.

I am using `kakoune` built from AUR: https://aur.archlinux.org/packages/kakoune-git

**Comment by occivink** [comment]
That's because there are two levels of expansions involved, once by `new` and then by `edit`. You can do 
`new edit a\\ b` or `new edit "a\ b"`, altough it's not super practical.

**Comment by gnull** [comment]
Now it's clear. Thank you.

**Comment by gnull** [comment]
@occivink, wait a minute. But the completion suggested for `:new edit` was incorrect. Shouldn't it be aware of multiple levels of expansion? Is this an issue?

**Comment by mawww** [comment]
Yeah, actually I think there should be no need to multiple escaping here, we should be able to "perfectly forward" arguments from one command to the next.

So yeah, there is an issue here, not sure yet how to fix it.

**Comment by crocket** [comment]
I am also facing this issue.

**Comment by occivink** [comment]
There is now the `-verbatim` flag of eval that can be used for perfect-forwarding within kakoune, but in this particular case (the eval string being specified with the `-e` CLI argument) it's not an option. 
It is a tricky case, but I think it could be made (hacked?) to work by reconstituting the argument split in the `new` command, instead of just joining them together as-is with `"%arg{@}"`

**Comment by alexherbo2** [comment]
It could be nice for `evaluate-commands` and `kak` to support rest arguments, similarly to `sh -c 'echo "$@"' -- a b c`.
